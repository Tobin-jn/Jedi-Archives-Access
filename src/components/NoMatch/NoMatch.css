.bad-path-message {
  text-align: center;
  color: #ADD8E6;
  font-family: "Lucida Console", Monaco, monospace;
  margin: 1rem;
}