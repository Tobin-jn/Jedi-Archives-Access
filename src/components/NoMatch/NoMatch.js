import React from 'react';

import './NoMatch.css';

const NoMatch = ({ location }) => {
  return(
    <section className='no-match' aria-label='404-error'>
      <h3 className='bad-path-message'>There is no entry within the archives for {location.pathname}</h3>
    </section>
  );
} 

export default NoMatch;