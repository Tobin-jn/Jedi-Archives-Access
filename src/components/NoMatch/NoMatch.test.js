/* eslint-disable */

import React from 'react';
import NoMatch from './NoMatch';
import { shallow } from 'enzyme';

let wrapper;
let mockLocation = {pathname: '/badpath'}
const errorMessage = [ 'There is no entry within the archives for ', '/badpath' ]

beforeEach(() => {
  wrapper = shallow(<NoMatch 
    location={mockLocation}
  />);
});

describe('NoMatch', () => {
  it ('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should receive location as props', () => {
    expect(wrapper.props().children.props.children).toEqual(errorMessage);
  });

});