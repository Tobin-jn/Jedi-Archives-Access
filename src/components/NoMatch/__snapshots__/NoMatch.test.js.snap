// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch should match the snapshot 1`] = `
<section
  aria-label="404-error"
  className="no-match"
>
  <h3
    className="bad-path-message"
  >
    There is no entry within the archives for 
    /badpath
  </h3>
</section>
`;
